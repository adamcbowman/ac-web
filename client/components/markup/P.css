@import '../../styles/colors.css';

.Capped {
    cursor: pointer;
    position: relative;
    margin-bottom: 1.25em;

    &:before {
        content: 'More';
        position: absolute;
        top: auto;
        right: 0;
        bottom: -1.25em;
        left: 0;
        text-align: right;
        color: var(--primary);
    }

    &:after {
        content: 'â€¦';
    }
}
