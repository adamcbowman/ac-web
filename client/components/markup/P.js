import React from 'react'
import { compose, withHandlers, withState } from 'recompose'
import PropTypes from 'prop-types'
import styles from './P.css'

P.propTypes = {
    children: PropTypes.string.isRequired,
    capAt: PropTypes.number,
    title: PropTypes.string,
}

function P({ children = '', capAt = Infinity, title, ...props }) {
    const capped = children.length > capAt
    const className = capped ? 'Capped' : null

    title = capped ? 'Click to read more' : title
    children = capped ? children.substr(0, capAt) : children

    return (
        <p className={styles[className]} {...props}>
            {children}
        </p>
    )
}

export default compose(
    withState('capAt', 'setCapAt', props => props.capAt),
    withHandlers({
        onClick: props => event => {
            event.preventDefault()

            props.setCapAt(Infinity)
        },
    })
)(P)
