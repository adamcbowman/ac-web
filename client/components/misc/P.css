@import "../../styles/colors.css";

.Capped {
    cursor: pointer;
    position: relative;
    margin-bottom: 1.25em;

    &:before {
        content: "More";
        position: absolute auto 0 -1.25em 0;
        text-align: right;
        color: var(--primary);
    }
    &:after {
        content: "â€¦";
    }
}
