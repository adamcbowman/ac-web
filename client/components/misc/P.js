import { DOM } from 'react'
import PropTypes from 'prop-types'
import CSSModules from 'react-css-modules'
import {
    compose,
    withHandlers,
    mapProps,
    withState,
    defaultProps,
    setPropTypes,
} from 'recompose'
import styles from './P.css'

export default compose(
    setPropTypes({
        children: PropTypes.string.isRequired,
        capAt: PropTypes.number,
    }),
    defaultProps({
        children: '',
        capAt: Infinity,
    }),
    withState('capAt', 'setCapAt', props => props.capAt),
    mapProps(({ capAt, children, title, ...props }) => {
        const capped = children.length > capAt

        return {
            ...props,
            children: capped ? children.substr(0, capAt) : children,
            styleName: capped ? 'Capped' : null,
            title: capped ? 'Click to read more' : title,
        }
    }),
    withHandlers({
        onClick: props => event => {
            event.preventDefault()

            props.setCapAt(Infinity)
        },
    }),
    CSSModules(styles)
)(DOM.p)
