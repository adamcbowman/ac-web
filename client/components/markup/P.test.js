import React from 'react'
import Renderer from 'react-test-renderer'
import P from './P'

test('p component', () => {
    expect(Renderer.create(<P>Some content...</P>)).toMatchSnapshot()
    expect(
        Renderer.create(<P capAt={10}>Some content truncated...</P>)
    ).toMatchSnapshot()
})
