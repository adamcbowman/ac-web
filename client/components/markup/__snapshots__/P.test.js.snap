// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`p component 1`] = `
<p
  onClick={[Function]}
  setCapAt={[Function]}
>
  Some content...
</p>
`;

exports[`p component 2`] = `
<p
  className="Capped"
  onClick={[Function]}
  setCapAt={[Function]}
>
  Some conte
</p>
`;
